firefox_puppeteer/tests/test_software_update.py

```diff
@@ -0,0 +1,108 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+from firefox_puppeteer.api.software_update import SoftwareUpdate
+
+
+class TestSoftwareUpdate(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+        self.software_update = SoftwareUpdate(self.get_marionette)
+        self.saved_mar_channels = self.software_update.mar_channels.channels
+        self.software_update.mar_channels.channels = 'expected,channels'
+
+    def tearDown(self):
+        self.software_update.mar_channels.channels = self.saved_mar_channels
+        FirefoxTestCase.tearDown(self)
+
+    def test_abi(self):
+        self.assertTrue(self.software_update.ABI)
+
+    def test_allowed(self):
+        self.assertEqual(True, self.software_update.allowed)
+
+    def test_build_info(self):
+        build_info = self.software_update.build_info
+        self.assertEqual(None, build_info['disabled_addons'])
+        self.assertIn('Mozilla/', build_info['user_agent'])
+        self.assertEqual(build_info['mar_channels'], ['expected', 'channels'])
+        self.assertTrue(build_info['version'])
+        self.assertTrue(build_info['buildid'].isdigit())
+        self.assertTrue(build_info['locale'])
+        self.assertIn('force=1', build_info['url_aus'])
+        self.assertEqual(self.software_update.update_channel.channel, build_info['channel'])
+
+    def test_force_fallback(self):
+        self.software_update.force_fallback()
+        with open(os.path.join(self.software_update.staging_directory, 'update.status'), 'r') as f:
+            content = f.read()
+        self.assertEqual(content, 'failed: 6\n')
+
+    def test_os_version(self):
+        self.assertTrue(self.software_update.os_version)
+
+    def test_staging_directory(self):
+        self.assertTrue(self.software_update.staging_directory)
+
+    def test_get_update_url(self):
+        self.assertIn('Firefox', self.software_update.get_update_url())
+        self.assertIn('Firefox', self.software_update.get_update_url(True))
+        self.assertIn('force=1', self.software_update.get_update_url(True))
```

Comment:
Maybe caching the call to `get_update_url()`? Should we also add `assertNotIn()` for `force=1` if not forced?